### PR TITLE
Improve bitski sdk api

### PR DESCRIPTION
**Pull request by pixelmatrix**
This is a proposal for an improved API to coincide with our final 0.2.0 release. The focus is on simplifying integration.

**New**

- getAuthStatus(): Determine whether or not you are already logged in, or if you can log in silently

- start(): Sign in silently if possible, otherwise sign in with the popup. Essential for a “Continue” flow.

- connect(): Sign in silently. Will fail if not possible. Useful for people who want to use a continuous session.

- signInRedirect(url): Sign in with the redirect flow, with the option to pass a different redirect url. The param is useful in scenarios where you support redirect flow and silent flow.

**Refined**

- signIn(): More opinionated sign in method leaves less decisions to make. Signs in with the popup method.

- getConnectButton(opts, callback): Use an options hash instead of a bunch of optional positional arguments.

- getUser(): Instead of attempting to log in, this method is now purely for getting user info when available.

- callback(): Callback method for popup and silent flow. Hopefully not used very often, but more clear that this is the default.

- redirectCallback(): Callback method for redirect flow. Explicitly for the redirect flow. Separate because we can’t as easily determine this use case.

- Bitski(clientId, redirectUrl, options): Removed the logout url, since we don’t use that oidc flow. In general pushing to use more option objects for pro features.

### CODE REVIEW

**Comment by pixelmatrix** [comment]
See README.md for examples of how this composes.

File: docs/README.md [hunk 1]
```diff
@@ -13,6 +13,42 @@ This repo consists of 2 packages:
...
+const bitski = new Bitski('CLIENT-ID', 'https://myapp.com/callback.html');
+
+const provider = bitski.getProvider();
+const web3 = new Web3(provider);
+
+// public calls are always available
+const network = await web3.eth.getBlockNumber();
+
+// connect via oauth to use the wallet
+await bitski.start();
```
Comment (ptescher) [inline docs/README.md]: Does this need a user action to call?

Comment (ptescher) [reply docs/README.md]: Might want to put that in a comment.

Comment (pixelmatrix) [reply docs/README.md]: In some cases yes, but this example is more pseudo code than copy / paste. The reality is that no ones code looks remotely like this and demonstrating a click handler here makes it much harder to follow.

**Comment by pixelmatrix** [comment]
Fixed the examples and found a couple of bugs

**Pull request merged**